operatorcourier/validate.py

```diff
@@ -184,79 +184,180 @@ def _csv_spec_validation(self, spec, bundleData):
             valid = False
 
         if "customresourcedefinitions" in spec:
-            customresourcedefinitions = spec["customresourcedefinitions"]
+            if self._csv_crd_validation(
+                    spec["customresourcedefinitions"], bundleData) is False:
+                valid = False
+
+        if "apiservicedefinitions" in spec:
+            if self._csv_asd_validation(spec["apiservicedefinitions"]) is False:
+                valid = False
+
+        return valid
+
+    def _csv_crd_validation(self, customresourcedefinitions, bundleData):
+        valid = True
+
+        crdList = []
+        for crd in bundleData[self.crdKey]:
+            try:
+                name = crd["metadata"]["name"]
+                crdList.append(name)
+            except KeyError:
+                pass
```

Comment:
Could this log a warning or error informing the user that a CRD is missing the `metadata.name` field? It's better to be noisy than quiet in this case (unless we previously warn elsewhere)